syntax = "proto3";

package protobuf;

option java_package = "de.uni_passau.fim.padas.padre.api.converter.protobuf";
option java_outer_classname = "DatasetV1Protos";

message DataFrameMeta {
  repeated string headers = 1;
}

message DataFrameRow {
  repeated DataFrameCell cells = 1;

  message DataFrameCell {
    oneof cell_type {		// Java		Python
      int32 int32_t = 1; 	// int		int
      int64 int64_t = 2;	// long		int/long
      double double_t = 3;	// double	float
      float float_t = 4; 	// float	float
      string string_t = 5;	// String	str/unicode
      bool bool_t = 6;	 	// boolean	bool
    }
  }
}
